### PR TITLE
Added several XSS rules

### PR DESCRIPTION
**Pull request by zaichenko**
Hello! 
I added several warnings about XSS rules and unit test XssInWebForms

### CODE REVIEW

File: SecurityCodeScan/Config/Main.yml [hunk 1]
```diff
@@ -3420,6 +3525,8 @@ Behavior:
     Method:
       Returns:
         Taint: Tainted
+    Field:
+      Injectable: [SCS0029: HtmlEscaped]
```
Comment (JarLob) [inline SecurityCodeScan/Config/Main.yml]: My test shows it is automatically escaped.

File: SecurityCodeScan/Config/Main.yml [hunk 2]
```diff
@@ -3444,6 +3553,8 @@ Behavior:
...
```
Comment (JarLob) [inline SecurityCodeScan/Config/Main.yml]: It is automatically escaped.

File: SecurityCodeScan/Config/Main.yml [hunk 3]
```diff
@@ -3452,6 +3563,8 @@ Behavior:
...
```
Comment (JarLob) [inline SecurityCodeScan/Config/Main.yml]: Not injectable, automatically escaped.

**Comment by JarLob** [comment]
Great new sinks! Created an issues for AppVeyor, for some reason the build is failing: https://help.appveyor.com/discussions/problems/18575-build-failed-sudden-jump-from-3-minutes-to-over-60

**Comment by zaichenko** [comment]
> Shouldn't it be UrlEncoded?

Will I have to add "UrlEncoded" to TaintTypes and it'll be enough?
I couldn't find how the check of using the validator (for example for HtmlEscaped) is implemented.

> My tests show that it is automatically html escaped (encoded) by the control.

I've somewhere read that it is possible to avoid encoded by default. I may be mistaken. I will remove it until I find where I've read it.

**Comment by JarLob** [comment]
The test checks if an exception is thrown when custom taint type limit is reached. Taint type names are represented as numbers. Since you have added built-in type you need to reduce the number of custom/per-user taint types by one.

**Comment by zaichenko** [comment]
I apologize for not answering for a long time. Holidays just recently ended. I corrected SanitizerTypesValidation test.

**Comment by JarLob** [comment]
I was quite busy too :)

I went through the list again and tried to make a POC for all sinks, but found some I couldn't exploit so here are my changes:
1. I have removed `Page.Title`. It is a sink I have added some time ago, found it somewhere. However it seems to be properly auto encoded in my tests.
2. Removed list based sinks. I'm less confident about this one. The issue is after I assign `Text` property with a payload it still doesn't render it in html. Maybe there is a way to trigger it. Removing for now.
3. I couldn't exploit `ImageUrl`. While user injection in `src=` is still bad and may be used for data exfiltration or CSRF (only if the server in url changes a state on GET) it is not XSS. I have moved these sinks to the audit configuration.
4. I played with `HyperLink.NavigateUrl` and noticed that `UrlEncode` actually breaks legal links like `http://www.microsoft.com`. Apparently the property does auto encoding and there is no trivial XSS except `javascript:alert(1)`. OWASP has quite complex rules for href (4th row): https://www.owasp.org/index.php/XSS_(Cross_Site_Scripting)_Prevention_Cheat_Sheet#XSS_Prevention_Rules_Summary but I found that `Microsoft.Security.Application.Encoder.UrlPathEncode` properly encodes both cases, so I introduced yet another taint type.

Let me know what you think and is it good to merge.

**Pull request merged**